Added pecs lowercase

josiebb opened the pull request:
pecs lowercase is just pecs but with two changes:

*reformatted the code with snake casing instead of camel casing

*changed the api and other internal function names to be shorter:
 -createECSworld -> create_world
 -createComponent -> create_comp
 -createEntity -> create_ent
 -createSystem -> create_sys

this was done with the intention of making it easier to use and read in the pico-8 code editor.

josiebb committed:
Added pecs lowercase

pecs lowercase is just pecs but with two changes:

*reformatted the code with snake casing instead of camel casing

*changed the api and other internal function names to be shorter:
 -createECSworld -> create_world
 -createComponent -> create_comp
 -createEntity -> create_ent
 -createSystem -> create_sys

this was done with the intention of making it easier to use and read in the pico-8 code editor.

josiebb committed:
fixed typo

jesstelford commented:
This looks good, thank you for the PR!

I'm tempted to make this the default for both pecs & pecs-lite, but I need to investigate the impact when I have some more time.

josiebb committed:
shaved some tokens

i was poking around for possible issues bc i had a memory problem in a project, which turned out to be in the project code after all. Still, i managed to shave off a handful of tokens and made some other, slightly arbitrary edits.

*shortened some single-condition/task if statements, which can exclude "then" and "end" and still work

*removed brackets around conditions for multi-line if statements, which will work without them.

*changed != to ~= because the editor I'm using seems unsettled by the exclamation points for some reason. Just highlights them in red, though the code works fine either way.

jesstelford commented:
Thank you @josiebb - I finally got some time to look at these changes (10 months later 😅), and I really liked them!

I took it a step further and changed the API to use single words, which reduces the character count a bit and removes the need for an underscore.

I've combined our changes and merged them in #2: See that PR for the specifics.

josiebb commented:
Oh wonderful! Glad you liked the changes i made. Your changes to the API
make sense. I have a project I've been working on a while with my edit of
PECS, i may switch it over to this version since the names are more
descriptive.

On Mon, Apr 18, 2022, 9:27 PM Jess Telford ***@***.***> wrote:

> Thank you @josiebb <https://github.com/josiebb> - I finally got some time
> to look at these changes (10 months later 😅), and I really liked them!
>
> I took it a step further and changed the API to use single words, which
> reduces the character count a bit and removes the need for an underscore.
>
> I've combined our changes and merged them in #2
> <https://github.com/jesstelford/pecs/pull/2>: See that PR for the
> specifics.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jesstelford/pecs/pull/1#issuecomment-1101904172>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASAMXUWH7G63HVICGF7DM7DVFYDZJANCNFSM467OQBKA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
